### PR TITLE
[examples] Fix HexahedronForceFieldTopologyChangeHandling scene

### PR DESCRIPTION
**Pull request by fredroy**
- Fix bad behavior of GridTopology

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

**Comment by fredroy** [comment]
[ci-build][with-all-tests]

File: examples/Components/topology/TopologicalModifiers/HexahedronForceFieldTopologyChangeHandling.scn [hunk 1]
```diff
@@ -7,13 +7,17 @@
     <DefaultContactManager response="default" />
     <MinProximityIntersection name="Proximity" alarmDistance="0.8" contactDistance="0.5" />
 
-    <RegularGridTopology name="grid" n="6 6 6" min="-10 -10 -10" max="10 10 10" p0="-10 -10 -10" />
+    <Node name="Grid">
+        <RegularGridTopology name="grid" n="6 6 6" min="-10 -10 -10" max="10 10 10" p0="-10 -10 -10" />
+        <MechanicalObject template="Vec3d" name="container" rest_position="@../grid.position"/>
```
Comment (epernod) [inline examples/Components/topology/TopologicalModifiers/HexahedronForceFieldTopologyChangeHandling.scn]: What does it change?

Comment (epernod) [reply examples/Components/topology/TopologicalModifiers/HexahedronForceFieldTopologyChangeHandling.scn]: I thought grid where supposed to work similarly as loaders

Comment (fredroy) [reply examples/Components/topology/TopologicalModifiers/HexahedronForceFieldTopologyChangeHandling.scn]: I dont really know, but randomly the test crashes and it seems the problem comes from data updates.
But GridTopology (and its siblings) is really weird and I dont think it behaves as a loader itself unfortunately.

**Review by epernod — APPROVED** [review]

**Pull request merged**